lib/chef_handler_foreman/foreman_facts.rb

```diff
@@ -67,8 +82,16 @@ def plainify(hash, prefix = nil)
           result.push plainify(array_to_hash(value), get_key(key, prefix))
         else
           new                       = {}
-          new[get_key(key, prefix)] = value
-          result.push new
+          full_key = get_key(key, prefix)
+          if @blacklist
```

Comment:
I need to check if the blacklist key is a substring of the full key, changed it to
```ruby
if @blacklist.nil? || !@blacklist.any? { |black_key| full_key.include?(black_key) }
  new[full_key] = value
  result.push new
end
```